documentation/src/main/asciidoc/topics/con_file_based_cache_stores.adoc

```diff
@@ -0,0 +1,77 @@
+[id='file-stores_{context}']
+= File-Based Cache Stores
+
+File-based cache stores provide persistent storage on the local host filesystem where {brandname} is running.
+For clustered caches, file-based cache stores are unique to each {brandname} node.
+
+You should use file-based cache stores to persist data that overflows from memory when the data container exceeds size or time restrictions.
+
+[WARNING]
+====
+Never use filesystem-based cache stores on shared file systems, such as an NFS or Samba share, because they do not provide file locking capabilities and data corruption can occur.
+
+Additionally if you attempt to use transactional caches with shared file systems, unrecoverable failures can happen when writing to files during the commit phase.
+====
+
+[discrete]
+== Soft-Index File Stores
+
+[NOTE]
+====
+`SoftIndexFileStore` is the default implementation for file-based cache stores.
+====
+
+`SoftIndexFileStore` is a Java implementation that uses a variant of **B+
+Tree** that is cached in-memory using Java soft references.
+The **B+ Tree**, called `Index` is offloaded on the file system to a single file that is purged and rebuilt each time the cache store restarts.
+
+`SoftIndexFileStore` stores data in a set of files rather than a single file.
+When usage of any file drops below 50%, the entries in the file are overwritten to another file and the file is then deleted.
+
+`SoftIndexFileStore` persists data in a set of files that are written in an append-only method.
+For this reason, if you use `SoftIndexFileStore` on conventional magnetic disk, it does not need to seek when writing a burst of entries.
+
+Most structures in `SoftIndexFileStore` are bounded, so out-of-memory exceptions do not pose a risk.
```

Comment:
Out of memory could still happen even with bounded structures if they are misconfigured. The main reason that this store will not pose out of memory issues is because the index is stored using java SoftReference as mentioned above so if it grows too big it will free those, while still falling back to the disk.